modules/javafx.controls/src/main/java/javafx/scene/control/Control.java

```diff
@@ -237,19 +237,6 @@ private static Class<?> loadClass(final String className, final Object instance)
         // a reference to the old value.
         private Skin<?> oldValue;
 
-        @Override
-        //This code is basically a kind of optimization that prevents a Skin that is equal but not instance equal.
```

Comment:
Why was this code here in the first place, considering that switching skins is surely not something that would logically be done repeatedly? Are there performance implications with this change?

Reply:
I don't know. I think someone guessed that if a new skin is from the same class it must be the same and can therefore be rejected. While this will probably work 95% the times, it is still not a 100% correct assumption.

This has no performance implications.